Lib9c/Action/RetrieveAvatarAssets.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Action.State;
+using Libplanet.Crypto;
+using Nekoyume.Model.State;
+using Nekoyume.Module;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType(TypeIdentifier)]
+    public class RetrieveAvatarAssets: ActionBase
+    {
+        public const string TypeIdentifier = "retrieve_avatar_assets";
+        public Address AvatarAddress;
+
+        public RetrieveAvatarAssets()
+        {
+        }
+
+        public RetrieveAvatarAssets(Address avatarAddress)
+        {
+            AvatarAddress = avatarAddress;
+        }
+
+        public override IValue PlainValue => Dictionary.Empty
+            .Add("type_id", TypeIdentifier)
+            .Add("values", Dictionary.Empty.Add("a", AvatarAddress.Serialize()));
+
+        public override void LoadPlainValue(IValue plainValue)
+        {
+            var asDict = (Dictionary)((Dictionary)plainValue)["values"];
+            AvatarAddress = asDict["a"].ToAddress();
+        }
+
+        public override IWorld Execute(IActionContext context)
+        {
+            context.UseGas(1);
+            Address signer = context.Signer;
+            var state = context.PreviousState;
+            if (state.TryGetAvatarState(signer, AvatarAddress, out _))
```

Comment:
If this line's purpose is only to check `AvatarAddress` is the address of `context.Signer`'s avatar, `TryGetAvatarState()` method can be too heavy. It should deserialize AvatarState from raw bencodex. It may affect to this action's performance.

Reply:
it would be better to just call GetAgentState and then check the `agentState.AvatarAddresses` 🤔